### ISSUE TITLE
Laravel 9.x w/ PHP 8 Composer Installation Failed

### ISSUE THREAD

**Issue by e2corporation** [UpCloudLtd/upcloud-php-api#53]
Installation of `upcloudltd/upcloud-php-api` via **Composer** fails with PHP 8.1.8.
```bash
$> php composer.phar require upcloudltd/upcloud-php-api --with-all-dependencies

```

Error Output
```bash
Running composer update upcloudltd/upcloud-php-api --with-all-dependencies
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - upcloudltd/upcloud-php-api[v2.1.0, ..., v2.3.0] require symfony/serializer ^5.1 -> satisfiable by symfony/serializer[v5.1.0, ..., v5.4.25].
    - symfony/mime[v6.2.0, ..., v6.3.0] conflict with symfony/serializer <6.2.
    - laravel/framework v9.52.10 requires symfony/mime ^6.0 -> satisfiable by symfony/mime[v6.0.0, ..., v6.3.0].
    - upcloudltd/upcloud-php-api v1.0.0 requires guzzlehttp/guzzle ^6.2 -> found guzzlehttp/guzzle[6.2.0, ..., 6.5.8] but it conflicts with your root composer.json require (^7.2).
    - upcloudltd/upcloud-php-api[v1.0.1-stable, ..., v1.0.3] require guzzlehttp/guzzle ^6.3 -> found guzzlehttp/guzzle[6.3.0, ..., 6.5.8] but it conflicts with your root composer.json require (^7.2).
    - upcloudltd/upcloud-php-api v2.0.0 requires php ^7.2 -> your php version (8.1.8) does not satisfy that requirement.
    - laravel/framework is locked to version v9.52.10 and an update of this package was not requested.
    - Root composer.json requires upcloudltd/upcloud-php-api * -> satisfiable by upcloudltd/upcloud-php-api[v1.0.0, v1.0.1-stable, v1.0.2, v1.0.3, v2.0.0, v2.1.0, v2.2.0, v2.3.0].

You can also try re-running composer require with an explicit version constraint, e.g. "composer require upcloudltd/upcloud-php-api:*" to figure out if any version is installable, or "composer require upcloudltd/upcloud-php-api:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

```

Are there any plans for `upcloudltd/upcloud-php-api` to support **PHP 8.x** on the roadmap?

**Comment by Darep** [comment]
Hi! Did you get this solved already? :blush: upcloud-php-api supports PHP 8.1 starting from v2.3.0. It would look to me like the `symfony/serializer` package is causing a version conflict between laravel/framework and upcloud-php-api and that causes the problem here. 🤔 Maybe the symfony/serializer version requirement in upcloud-php-api should be relaxed a little bit…

**Comment by kangasta** [comment]
Hello, this should now be resolved in [v3.0.0](https://github.com/UpCloudLtd/upcloud-php-api/releases/tag/v3.0.0).

**Comment by kangasta** [comment]
I'll close this as completed, please re-open or create a new issue, if there are still issues when installing together with Laravel.
